byway/src/main/webapp/home.js

```diff
@@ -110,7 +148,14 @@ function calcRoute(directionsService, directionsRenderer, start, end, waypoints)
   directionsService.route(request, function(response, status) {
     if (status == 'OK') {
       directionsRenderer.setDirections(response);
-    } else {
+    } 
+    else if (status === google.maps.DirectionsStatus.OVER_QUERY_LIMIT) {
+      delay++;
+      setTimeout(function () {
+        calcRoute(directionsService,directionsRenderer, start, end, waypoints);
+      }, delay * 1000);
```

Comment:
Might want to use delayPromise function from recommendations.js, you can move into base.js and align/sync in html/js files as needed for eslint errors

Reply:
okay changed to use delayPromises. For now I am keeping the delayPromises function in both home.js and recomendations.js until we decide what to do with base.js and utils.js